app/actions/remote/entry/app.ts

```diff
@@ -1,23 +1,41 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
+import {resetAfterCRTChange} from '@actions/app/global';
 import {switchToChannelById} from '@actions/remote/channel';
 import {fetchConfigAndLicense} from '@actions/remote/systems';
+import {queryHasCRTChanged} from '@app/queries/servers/preference';
 import DatabaseManager from '@database/manager';
 import {prepareCommonSystemValues, getCommonSystemValues, getCurrentTeamId, getWebSocketLastDisconnected, setCurrentTeamAndChannelId, getCurrentChannelId} from '@queries/servers/system';
 import {getCurrentUser} from '@queries/servers/user';
 import {deleteV1Data} from '@utils/file';
 import {isTablet} from '@utils/helpers';
 
+import {fetchMyPreferences} from '../preference';
+
 import {deferredAppEntryActions, entry, registerDeviceToken, syncOtherServers, verifyPushProxy} from './common';
 
-export async function appEntry(serverUrl: string, since = 0, isUpgrade = false) {
+export async function appEntry(serverUrl: string, since = 0, isUpgrade = false): Promise<{error?: any; userId?: string}> {
     const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
     if (!operator) {
         return {error: `${serverUrl} database not found`};
     }
 
     const {database} = operator;
+    const myPreferences = await fetchMyPreferences(serverUrl, true);
```

Comment:
I need to do it here though, I'll see if it can be removed from `fetchEntryData`

Reply:
`appEntry` calls `entry` and other entry functions, so we need to call `appEntry` upon changing CRT status.
And furthermore we "need" to "stop" execution at that point so we don't fetch data 2 times.

Reply:
I don't think you need to fetch data twice.

You could just check in `entry` if the flag was toggled and delete the tables data before continuing with the existing flow

Reply:
My issue with that is that if we change the order of execution in `appEntry` it will break.
Making it on top, and calling `appEntry` again is clearer on intent IMO, 0/5 though

Reply:
Clearer on intent could be, I find it a bit uneccessary, as I don't see the break in the execution that you mention but maybe I'm missibg something.

Also the fact that we are fething and processing the preferences twice (hunders if them).

The way I see it is this:
1. On app open:
    - app entry is called
    - entry fetches data (including prefences)
    - in the entry function we check for the toggle
    - if toggled, truncate the tables
    - continue with the rest of the exising code
2. Opening from push notification
   - notification entry is called
   - same as 1
3. Preference change WS event
   - check for toggle
   - truncate tables
   - call appEntry

Reply:
Still we'll need to decouple fetching my preferences away from `fetchAppEntryData`.
e.g: we need to truncate tables before fetching my channel data.

Having that check in the top of `appEntry` and calling `appEntry` again, in case CRT has been toggled, seems way much more clear to me.

Reply:
@enahum what if we just remove it from `fetchAppEntryData`?

Reply:
And pass preferences as arguments